pkg/collector/horizontalpodautoscaler.go

```diff
@@ -26,30 +29,78 @@ import (
 
 const defaultCPUTarget int32 = 90
 
-func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) autoscalingv1.HorizontalPodAutoscaler {
+func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) runtime.Object {
```

Comment:
instread of returning `runtime.Object` wouldn't it be better to return `client.Object`? That way we don't have to cast this in the reconcile.